### PR TITLE
Add rule retriever yaml test with empty indices

### PR DESCRIPTION
**Pull request by kderusso**
See: SEARCH-910

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java [hunk 1]
```diff
@@ -30,6 +30,7 @@ public class InferenceFeatures implements FeatureSpecification {
     private static final NodeFeature TEST_RERANKING_SERVICE_PARSE_TEXT_AS_SCORE = new NodeFeature(
         "test_reranking_service.parse_text_as_score"
     );
+    private static final NodeFeature TEST_RULE_RETRIEVER_WITH_EMPTY_INDEX = new NodeFeature("test_rule_retriever.with_empty_index");
```
Comment (kderusso) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java:33]: I couldn't find the feature this was fixed in, so I added a new test feature.

**Review by Samiul-TheSoccerFan — APPROVED** [review]
LGTM

**Review by mridula-s109 — APPROVED** [review]

**Review by Mikep86 — APPROVED** [review]
One nit comment about names. Also, should we backport this to 8.19?

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kderusso) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java:33]: I mean, technically it's empty in the result set, but I'll make a more verbose name 😜

[the remark this answers is not in the document]
Comment (Mikep86) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/InferenceFeatures.java:33]: "Empty result set" is more descriptive, thank you :)
